src/utils/domUtils.ts

```diff
@@ -107,6 +107,15 @@ export module _Util {
       if (boxA.top > boxB.bottom) { return false; }
       return true;
     }
+
+    export function boxIsInside(inner: ClientRect, outer: ClientRect) {
+      return (
+        Math.floor(outer.left) <= Math.ceil(inner.left) &&
+        Math.floor(outer.top)  <= Math.ceil(inner.top)  &&
+        Math.floor(inner.right)  <= Math.ceil(outer.right) &&
+        Math.floor(inner.bottom) <= Math.ceil(outer.bottom)
+      );
+    }
```

Comment:
I wanted to cherry-pick from Roger's branch, but there wasn't a single commit that added just those lines. Now that I think about it, I should have done that...